plugins/catalog-backend-module-ldap/src/ldap/vendors.ts

```diff
@@ -39,48 +40,26 @@ export type LdapVendor = {
   decodeStringAttribute: (entry: SearchEntry, name: string) => string[];
 };
 
-export const DefaultLdapVendor: LdapVendor = {
-  dnAttributeName: 'entryDN',
-  uuidAttributeName: 'entryUUID',
-  decodeStringAttribute: (entry, name) => {
-    return decode(entry, name, value => {
-      return value.toString();
-    });
-  },
-};
-
-export const ActiveDirectoryVendor: LdapVendor = {
-  dnAttributeName: 'distinguishedName',
-  uuidAttributeName: 'objectGUID',
-  decodeStringAttribute: (entry, name) => {
-    const decoder = (value: string | Buffer) => {
-      if (name === ActiveDirectoryVendor.uuidAttributeName) {
-        return formatGUID(value);
-      }
-      return value.toString();
-    };
-    return decode(entry, name, decoder);
-  },
-};
-
-export const FreeIpaVendor: LdapVendor = {
-  dnAttributeName: 'dn',
-  uuidAttributeName: 'ipaUniqueID',
-  decodeStringAttribute: (entry, name) => {
-    return decode(entry, name, value => {
-      return value.toString();
-    });
-  },
-};
-
-export const AEDirVendor: LdapVendor = {
-  dnAttributeName: 'dn',
-  uuidAttributeName: 'entryUUID',
-  decodeStringAttribute: (entry, name) => {
-    return decode(entry, name, value => {
-      return value.toString();
-    });
-  },
+export const CreateLdapVendor = (
+  vendorConfig: VendorConfig,
+  isActiveDirectoryVendor: boolean,
+): LdapVendor => {
+  return {
+    dnAttributeName: vendorConfig.dnAttributeName || `entryDN`,
+    uuidAttributeName: vendorConfig.uuidAttributeName || `entryUUID`,
```

Comment:
You can't replace these defaults. Note that they were all different before in the code above. The whole point of all of this code was that you should NOT have to configure the dn/uuid because they get automagically detected for you. Plug-and-play, ideally. I suggest reverting this entire file